include/oneapi/dpl/internal/async_impl/async_backend_sycl.h

```diff
@@ -0,0 +1,131 @@
+// -*- C++ -*-
+//===-- async_backend_sycl.h ----------------------------------------------===//
```

Comment:
What's a goal of the file?
Do we need to add  special "async SYCL" backend? Why? The current SYCL backend is already asynchronous by essence.
And it additionally needs some minor changes like returning a future object (with events and/or temporary storages), nothing more.

Reply:
Parallel transform reduce pattern has an implicit synchronization point because it is returning the result by value.
Probably we don't need a special async SYCL backend if all changes would have been integrated to the existing one. Goal is to keep both pattern separate for the next release (async pattern as experimental feature)

Reply:
But you can drop the implicit synchronization point and return a future object. And a caller (on PSTL high level or on the highest level) makes 'wait'.
And I don't see  a significant reason to keep both patterns (and huge code duplication), even if it is as experimental feature.

Reply:
To tell the truth, I don't agree with that not perfect design and the huge code duplication may be accepted to merge in the main branch. 
Furthermore, the keeping  the same SYCL backend is pretty simple and it can we done in couple of days.

Reply:
I've created an issue: https://github.com/oneapi-src/oneDPL/issues/136

Reply:
I would not leave for later these changes. 

1. It looks strange  - to duplicate the same backend (up to 99%) code, merge it, and after - remove one.
2. #136 marked as Experimental  - that changes will not be done for ages :-)

Pablo,  I can do it (keep the same SYCL backend) in one day (on your branch)